common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -238,8 +243,78 @@ def student_view(self, context):
 
         fragment.add_content(self.system.render_template("seq_module.html", params))
 
+        self._capture_full_seq_item_metrics(display_items)
+        self._capture_current_unit_metrics(display_items)
+
+        # Get all descendant XBlock types and counts
         return fragment
 
+    def _locations_in_subtree(self, node):
+        """
+        The usage keys for all descendants of an XBlock/XModule as a flat list.
+        """
+        stack = [node]
+        locations = []
+
+        while stack:
+            curr = stack.pop()
+            locations.append(curr.location)
+            if curr.has_children:
+                stack.extend(curr.get_children())
```

Comment:
Children are [cached by default](https://github.com/edx/XBlock/blob/17956e7e85449e6e99c7ac33f56209b4d12846e4/xblock/mixins.py#L391), though it's entirely possible to implement a custom `get_children()` that is not. I think this should be ok.